driver/src/price_finding/optimization_price_finder.rs

```diff
@@ -3,98 +3,181 @@ use crate::price_finding::error::{ErrorKind, PriceFindingError};
+                .iter()
+                .map(|(token, price)| (token.0, price.unwrap_or_default().0))
+                .collect();
+            let executed_sell_amounts = self
+                .orders
+                .iter()
+                .map(|order| order.exec_sell_amount.0)
+                .collect();
+            let executed_buy_amounts = self
+                .orders
+                .iter()
+                .map(|order| order.exec_buy_amount.0)
+                .collect();
+
+            Solution {
+                prices,
+                executed_sell_amounts,
+                executed_buy_amounts,
+            }
+        }
     }
 }
 
 mod solver_input {
-    use serde::{Serialize, Serializer};
-    use std::collections::{BTreeMap, HashMap};
+    use super::{Num, TokenId};
+    use crate::models;
+    use crate::price_finding;
+    use ethcontract::H160;
+    use serde::Serialize;
+    use std::collections::{BTreeMap, BTreeSet};
     use std::vec::Vec;
 
+    /// Fee information using `TokenId` so the JSON serialization format matches
+    /// what is expected by the solver.
+    ///
+    /// This type may be removed if the `crate::price_finding::Fee` is converted
+    /// to use `TokenId` in the future.
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Fee {
+        pub token: TokenId,
+        pub ratio: f64,
+    }
+
+    impl From<&'_ price_finding::Fee> for Fee {
+        fn from(fee: &price_finding::Fee) -> Self {
+            Fee {
+                token: TokenId(fee.token),
+                ratio: fee.ratio,
+            }
+        }
+    }
+
+    /// Order information using `TokenId` so the JSON serialization format
+    /// matches what is expected by the solver.
+    ///
+    /// This type may be removed if the `crate::modes::Order` is converted to
+    /// use `TokenId` in the future.
     #[derive(Serialize)]
     #[serde(rename_all = "camelCase")]
     pub struct Order {
         #[serde(rename = "accountID")]
-        pub account_id: String,
-        pub sell_token: String,
-        pub buy_token: String,
-        pub sell_amount: String,
-        pub buy_amount: String,
+        pub account_id: H160,
+        pub sell_token: TokenId,
+        pub buy_token: TokenId,
+        pub sell_amount: Num,
+        pub buy_amount: Num,
     }
 
-    #[derive(Serialize)]
-    pub struct Fee {
-        pub token: String,
-        pub ratio: f64,
+    impl From<&'_ models::Order> for Order {
+        fn from(order: &models::Order) -> Self {
+            Order {
+                account_id: order.account_id,
+                sell_token: TokenId(order.sell_token),
+                buy_token: TokenId(order.buy_token),
+                sell_amount: Num(order.sell_amount),
+                buy_amount: Num(order.buy_amount),
+            }
+        }
     }
 
-    pub type Accounts = HashMap<String, HashMap<String, String>>;
+    pub type Accounts = BTreeMap<H160, BTreeMap<TokenId, Num>>;
```

Comment:
Whats the benefit of `BTree` instead of `HashMap`?

Reply:
`BTreeMap`s keep the key value pairs in a sorted (on key) binary tree, the cool property about this is that a depth first traversal of the tree will return the key value pairs sorted on key.

As we wanted to serialize the accounts such that they are alphabetically sorted on the account address, using a `BTreeMap` here just gives us that for free.